File: android/measure/src/main/java/sh/measure/android/bugreport/ShakeDetector.kt
package sh.measure.android.bugreport

/**
 * Interface defining shake detection capability
 */
internal interface ShakeDetector {
    /**
     * Start listening for shake events
     * @return true if successfully started, false otherwise
     */
    fun start(): Boolean

    /**
     * Stop listening for shake events
     */
    fun stop()

    /**
     * Set a listener to be notified of shake events
     * @param listener The listener to notify
     */
    fun setShakeListener(listener: Listener?)

    interface Listener {
        fun onShake()
    }
}

File: ios/Sources/MeasureSDK/Swift/BugReport/ShakeDetector.swift
//
//  ShakeDetector.swift
//  Measure
//
//  Created by Adwin Ross on 09/05/25.
//

import Foundation

/// Interface defining shake detection capability
protocol ShakeDetector {
    /// Start listening for shake events
    /// - Returns: `true` if successfully started, `false` otherwise
    @discardableResult
    func start() -> Bool

    /// Stop listening for shake events
    func stop()

    /// Set a listener to be notified of shake events
    func setShakeListener(_ listener: ShakeDetectorListener?)

    /// For testing: Get the current shake listener
    func getShakeListener() -> ShakeDetectorListener?
}

/// Listener protocol to receive shake events
protocol ShakeDetectorListener: AnyObject {
    func onShake()
}
